### PR TITLE
JAMES-2470 Use a single CassandraCluster per test class

### PR DESCRIPTION
**Pull request by chibenwa**
This way we can establish the connection only one time per test class.

Only the 3 last commits:

 - https://github.com/linagora/james-project/commit/ba964ad10d327b99347f6d309f5039a988a190f3 `
MAILBOX-342 Parallelize massive insertions `
 - https://github.com/linagora/james-project/commit/3e0c905c3e34bf54f8fd576697fddb4dada9fbd5 `JAMES-2470 MPT Cassandra should rely on CassandraCluster for cleanup`
 - https://github.com/linagora/james-project/commit/044c4192eb7c94a2703e92c516f40b4423909a5b `
JAMES-2470 Use a single Cassandra Cluster for a full test class `

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[6b45a6fb330475abc76026b7ea0679d97fdf9a8a] [ERROR] COMPILATION ERROR : 
[6b45a6fb330475abc76026b7ea0679d97fdf9a8a] [INFO] -------------------------------------------------------------
[6b45a6fb330475abc76026b7ea0679d97fdf9a8a] [ERROR] /james-project/mpt/impl/imap-mailbox/hbase/src/test/java/org/apache/james/mpt/imapmailbox/hbase/HBaseQuotaTest.java:[47,5] method does not override or implement a method from a supertype
[6b45a6fb330475abc76026b7ea0679d97fdf9a8a] [INFO] 1 error
```

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/event/distributed/CassandraBasedRegisteredDistributedMailboxDelegatingListenerTest.java [hunk 1]
```diff
@@ -80,6 +83,12 @@
     private EventCollector eventCollectorOnce3;
     private MailboxSession mailboxSession;
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/event/distributed/CassandraBasedRegisteredDistributedMailboxDelegatingListenerTest.java]: one extra line

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapperTest.java [hunk 2]
```diff
@@ -41,22 +41,30 @@
...
+
+    @ClassRule public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
+    private static CassandraCluster cassandra;
+
     private CassandraACLMapper cassandraACLMapper;
-    private CassandraCluster cassandra;
     private ExecutorService executor;
 
-    @ClassRule public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
+
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapperTest.java]: same here

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentDAOV2Test.java [hunk 3]
```diff
@@ -44,23 +46,29 @@
     @ClassRule
     public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
 
-    private CassandraCluster cassandra;
+    private static CassandraCluster cassandra;
 
     private CassandraAttachmentDAOV2 testee;
 
+
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentDAOV2Test.java]: same here

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentFallbackTest.java [hunk 4]
```diff
@@ -52,25 +54,25 @@
...
     private CassandraAttachmentDAOV2 attachmentDAOV2;
     private CassandraAttachmentDAO attachmentDAO;
     private CassandraAttachmentMapper attachmentMapper;
     private CassandraBlobsDAO blobsDAO;
     private CassandraAttachmentMessageIdDAO attachmentMessageIdDAO;
 
-    @Before
-    public void setUp() throws Exception {
-        CassandraModuleComposite compositeModule = new CassandraModuleComposite(
+
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraAttachmentFallbackTest.java]: same here

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraGenericMailboxMapperTest.java [hunk 5]
```diff
@@ -32,30 +32,42 @@
...
+import org.junit.BeforeClass;
 import org.junit.ClassRule;
 
 public class CassandraGenericMailboxMapperTest extends MailboxMapperTest {
     
     @ClassRule public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
     
-    private CassandraCluster cassandra;
+    private static CassandraCluster cassandra;
+
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraGenericMailboxMapperTest.java]: same here

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdToImapUidDAOTest.java [hunk 6]
```diff
@@ -46,22 +48,33 @@
 public class CassandraMessageIdToImapUidDAOTest {
 
     @ClassRule public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
-    
-    private CassandraCluster cassandra;
+
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdToImapUidDAOTest.java]: same here

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentMessageIdCreationTest.java [hunk 7]
```diff
@@ -75,15 +79,18 @@
     private SimpleMailboxMessage message;
     private CassandraMessageId messageId;
 
+
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentMessageIdCreationTest.java]: same here

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentV2MigrationTest.java [hunk 8]
```diff
@@ -61,15 +65,17 @@
     private Attachment attachment1;
     private Attachment attachment2;
 
-    @Before
-    public void setUp() {
-        CassandraCluster cassandra = CassandraCluster.create(
-            new CassandraModuleComposite(
+
```
Comment (aduprat) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/migration/AttachmentV2MigrationTest.java]: same here

File: server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/CassandraSieveDAOTest.java [hunk 9]
```diff
@@ -60,19 +62,29 @@
         .build();
 
     @ClassRule public static DockerCassandraRule cassandraServer = new DockerCassandraRule();
-    
-    private CassandraCluster cassandra;
+    private static CassandraCluster cassandra;
+
     private CassandraSieveDAO sieveDAO;
 
+
```
Comment (aduprat) [inline server/data/data-cassandra/src/test/java/org/apache/james/sieve/cassandra/CassandraSieveDAOTest.java]: same here

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMessageIdManagerQuotaTest.java [hunk 10]
```diff
@@ -43,40 +44,52 @@
...
+            new CassandraFirstUnseenModule(),
+            new CassandraUidModule(),
+            new CassandraModSeqModule(),
+            new CassandraSubscriptionModule(),
+            new CassandraAttachmentModule(),
+            new CassandraDeletedMessageModule(),
+            new CassandraAnnotationModule(),
+            new CassandraApplicableFlagsModule(),
+            new CassandraQuotaModule());
+        cassandra = CassandraCluster.create(modules, cassandraServer.getIp(), cassandraServer.getBindingPort());
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMessageIdManagerQuotaTest.java]: no .getHost() here?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMessageIdManagerStorageTest.java [hunk 11]
```diff
@@ -43,39 +43,50 @@
...
+            new CassandraMailboxRecentsModule(),
+            new CassandraFirstUnseenModule(),
+            new CassandraDeletedMessageModule(),
+            new CassandraUidModule(),
+            new CassandraModSeqModule(),
+            new CassandraAttachmentModule(),
+            new CassandraAnnotationModule(),
+            new CassandraApplicableFlagsModule(),
+            new CassandraBlobModule());
+        cassandra = CassandraCluster.create(modules, cassandraServer.getIp(), cassandraServer.getBindingPort());
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraMessageIdManagerStorageTest.java]: idem

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraSubscriptionManagerTest.java [hunk 12]
```diff
@@ -59,22 +61,31 @@
...
-    public void init() {
+    private static CassandraCluster cassandra;
+
+    @BeforeClass
+    public static void setUpClass() {
         CassandraModuleComposite modules = new CassandraModuleComposite(
             new CassandraSubscriptionModule(),
             new CassandraMailboxCounterModule());
-        cassandra = CassandraCluster.create(modules, cassandraServer.getHost());
+        cassandra = CassandraCluster.create(modules, cassandraServer.getIp(), cassandraServer.getBindingPort());
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/CassandraSubscriptionManagerTest.java]: I don't understand why in this commit sometimes you replace by getHost, sometimes you remove getHost call.
That's a good example of a commit you should split. This is already a big commit, and you should not mix the initial commit goal with some little improvements like that.

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**